frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx

```diff
@@ -21,14 +21,15 @@ export const getSelectOverrides = (): MantineThemeOverride["components"] => ({
       },
     }),
     styles: (theme, _, { size = "md" }) => ({
-      ...getSelectInputOverrides(theme),
+      ...getSelectInputOverrides(theme, size),
       ...getSelectItemsOverrides(theme, size),
     }),
   },
 });
 
 export const getSelectInputOverrides = (
   theme: MantineTheme,
+  size: MantineSize | number,
```

Comment:
This component is causing some weird styling conflicts for selected and hovered states.

**Initial Open of Popover**
![Screenshot from 2023-12-28 17-11-58](https://github.com/metabase/metabase/assets/22608765/e54fa7dd-625f-4e39-bf56-46b53e0bfa71)

**Move mouse outside of main column button**
![Screenshot from 2023-12-28 17-12-03](https://github.com/metabase/metabase/assets/22608765/403f247f-bbbe-4c95-a096-5668e6c95daa)

**Hover over Item**
![Screenshot from 2023-12-28 17-19-57](https://github.com/metabase/metabase/assets/22608765/3d0339ca-626e-4062-aeb6-a47d72c07bae)

Reply:
Nice catch, although it seems to be a common behavior across all `Select` components ([check out Storybook](https://storybook.metabase.com/?path=/story/inputs-select--default-story) for example). Let's bring that up to the rest of the team

Reply:
Were you going to bring that up to the team, or would you prefer I do it?